### DISCUSSION TITLE
How to use Strongly type view model in template?

### DISCUSSION THREAD

**Discussion by Sushi21** [adoconnection/RazorEngineCore#50] [Q&A]
Hi,

Im looking forward to use a library like this in order to create some sort of PDF engine.

Im running few tests here and there.

I'm not sure what I'm doing wrong but I would like to be able to use strong type model within the template.

Am I doing something wrong?

Cheers

![image](https://user-images.githubusercontent.com/35455125/114094020-2ea2cb00-9892-11eb-95f3-2d22cf10f551.png)

**Comment by adoconnection** [comment]
Hi, sorry for super late reply.
Looks perfectly ok to me. Will it compile or it’s just intelligence glitch?

**Reply by garryxiao** [reply]
I would like to change a little bit about strong type model with the intelligence support:
During razorEngine.Compile: "M" is the generic type of the model.
                    // Add the model type reference
                    builder.AddAssemblyReference(typeof(M));
In the template, use:
var model = @Model as Tests.Web.TestModel;
Should work.

**Reply by jackhamby** [reply]
putting this here for reference. 
https://github.com/adoconnection/RazorEngineCore/issues/13
same issue happens on build for me. something to do with mvc razor precompiling templates. if you change the file extension you might get it to build (but not great intellisense wise)
